website/project/views/drafts.py

```diff
@@ -0,0 +1,210 @@
+# @must_be_valid_project
+# def submit_draft_for_review(auth, node, draft_pk, *args, **kwargs):
+#     user = auth.user
+#
+#     draft = get_draft_or_fail(draft_pk)
+#     draft.is_pending_review = True
+#     draft.save()
+#
+#     REVIEW_EMAIL = Mail(tpl_prefix='prereg_review', subject='New Prereg Prize registration ready for review')
+#     send_mail(draft.initiator.email, REVIEW_EMAIL, user=user, src=node)
+#
+#     ret = project_utils.serialize_node(node, auth)
+#     ret['success'] = True
+#     return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def draft_before_register_page(auth, node, draft, *args, **kwargs):
+    ret = serialize_node(node, auth, primary=True)
+
+    ret['draft'] = serialize_draft_registration(draft, auth)
+    return ret
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+@http_error_if_disk_saving_mode
+def register_draft_registration(auth, node, draft, *args, **kwargs):
+
+    data = request.get_json()
+    register = draft.register(auth)
+
+    try:
+        if data.get('registrationChoice', 'immediate') == 'embargo':
+            # Initiate embargo
+            embargo_end_date = parse_date(data['embargoEndDate'], ignoretz=True)
+            register.embargo_registration(auth.user, embargo_end_date)
+        else:
+            register.require_approval(auth.user)
+        register.save()
+    except ValidationValueError as err:
+        raise HTTPError(http.BAD_REQUEST, data=dict(message_long=err.message))
+
+    push_status_message(language.AFTER_REGISTER_ARCHIVING,
+                        kind='info',
+                        trust=False)
+    return {
+        'status': 'initiated',
+        'urls': {
+            'registrations': node.web_url_for('node_registrations')
+        }
+    }, http.ACCEPTED
+
+@autoload_draft
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registration(auth, node, draft, *args, **kwargs):
+    return serialize_draft_registration(draft, auth), http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def get_draft_registrations(auth, node, *args, **kwargs):
+
+    count = request.args.get('count', 100)
+    drafts = node.draft_registrations.find(
+        Q('registered_node', 'eq', None)
+    )[:count]
+    return {
+        'drafts': [serialize_draft_registration(d, auth) for d in drafts]
+    }, http.OK
+
+@must_have_permission(ADMIN)
+@must_be_valid_project
+def create_draft_registration(auth, node, *args, **kwargs):
+
+    data = request.get_json()
+
+    schema_name = data.get('schema_name')
+    if not schema_name:
+        raise HTTPError(http.BAD_REQUEST)
+
+    schema_version = data.get('schema_version', 1)
+    schema_data = data.get('schema_data', {})
+
+    meta_schema = get_schema_or_fail(
+        Q('name', 'eq', schema_name) &
+        Q('schema_version', 'eq', schema_version)
+    )
+    draft = node.create_draft_registration(auth.user, meta_schema, schema_data, save=True)
```

Comment:
Also, can we validate the submitted schema data against the actual schema, to be sure that all inputs are filled out correctly? Otherwise this could lead to malformed requests and user issues later.

We may need to look at validation separately due to time constraints, but it's particularly relevant if this ever becomes part of a public-facing api where people are constructing requests without the help of our frontend.